components/UpdateEmailForm/UpdateEmailForm.tsx

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Gridiron Survivor.
+// Licensed under the MIT License.
+
+'use client';
+import { updateUserEmail } from '@/api/apiFunctions';
+import Alert from '@/components/AlertNotification/AlertNotification';
+import { AlertVariants } from '@/components/AlertNotification/Alerts.enum';
+import { Button } from '@/components/Button/Button';
+import { Input } from '@/components/Input/Input';
+import LoadingSpinner from '@/components/LoadingSpinner/LoadingSpinner';
+import { useDataStore } from '@/store/dataStore';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { JSX, useState } from 'react';
+import { Control, SubmitHandler, useForm, useWatch } from 'react-hook-form';
+import toast from 'react-hot-toast';
+import { z } from 'zod';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '../Form/Form';
+
+const UpdateEmailSchema = z.object({
+  email: z
+    .string()
+    .min(1, { message: 'Please enter an email address' })
+    .email({ message: 'Please enter a valid email address' }),
+  password: z.string().min(1, { message: 'Please enter a password' }),
+});
+
+type UpdateEmailSchemaType = z.infer<typeof UpdateEmailSchema>;
+
+/**
+ * Display update user email form
+ * @returns {JSX.Element} The rendered update user email form.
+ */
+const UpdateEmailForm = (): JSX.Element => {
+  const [isUpdating, setIsUpdating] = useState<boolean>(false);
+  const { user, updateUser } = useDataStore((state) => state);
+
+  const form = useForm<UpdateEmailSchemaType>({
+    resolver: zodResolver(UpdateEmailSchema),
+    defaultValues: {
+      email: user.email,
+      password: '',
+    },
+  });
+
+  const email: string = useWatch({
+    control: form.control,
+    name: 'email',
+  });
+
+  const password: string = useWatch({
+    control: form.control,
+    name: 'password',
+  });
```

Comment:
Can you explain what we're using `useWatch` for with password and email?

Reply:
@choir27 basically useWatch is a react-hook-for.

Here's what useWatch is doing:

- It's watching the 'email' and 'password' fields of the form.
- It returns the current value of these fields.
- It will update these values whenever they change in the form, without causing a re-render of the entire component.